pkg/ocm/clusters.go

```diff
@@ -605,3 +606,25 @@ func (c *Client) ResumeCluster(clusterID string) error {
 
 	return nil
 }
+
+//Get all the archived ROSA STS clusters
+func (c *Client) GetArchivedCluster(clusterKey string) (clusters []*cmv1.Cluster, err error) {
+	query := fmt.Sprintf("id = '%s' OR name = '%s' OR external_id = '%s'",
+		clusterKey, clusterKey, clusterKey,
+	)
```

Comment:
This filter should be:
```go
	query := fmt.Sprintf("%s AND (id = '%s' OR name = '%s' OR external_id = '%s')",
		getClusterFilter(creator),
		clusterKey, clusterKey, clusterKey,
```
Otherwise we might end up fetching clusters installed in a different AWS account